docs/gene_sets.md

```diff
@@ -0,0 +1,103 @@
+
+The order of the gene symbols in each gene set is maintained in the database. 
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED" "MAY", and "OPTIONAL" in this document are to be interpreted as described in BCP 14, [RFC2119](https://www.rfc-editor.org/rfc/rfc2119.txt), and [RFC8174](https://www.rfc-editor.org/rfc/rfc8174.txt) when, and only when, they appear in all capitals, as shown here.
+
+## cellxgene CLI commands
+
+Users MAY use --user-generated-data-dir `file_name` to designate a file where both gene sets and annotations CSVs will be saved.
+
+Users MAY use --gene-sets-file `name_of_file.csv` to designate a CSV with preexisting gene sets you would like to view in cellxgene. The CSV MUST follow the format below.
+
+## cellxgene gene set CSV data format
+  
+The cellxgene gene set data format is a [*Tidy* CSV](./gene_sets_example.csv) (comma-separated values) file using ASCII encoding. Multiple gene sets MAY be included in the file similar to the [Gene Matrix Transposed](https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29) or [Gene Matrix](https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMX:_Gene_MatriX_file_format_.28.2A.gmx.29) formats.
+
+Example:
+
+| gene_set_name | gene_set_description | gene_symbol | gene_description | provenance1      | provenance1_description |
+|---------------|----------------------|-------------|------------------|------------------|-------------------------|
+| club.cell     | description          | CCKAR       | description      | Pubmed ID XYZ123 | Primary Pubmed ID       |
+| club.cell     |                      | SCGB3A2     | description      | Pubmed ID ABC456 | Primary Pubmed ID       |
+| club.cell     |                      | CYP2F2      | description      | Pubmed ID DCF678 | Primary Pubmed ID       |
+| macrophage    | description          | CD68        | description      |                  |                         |
+| macrophage    |                      | CD163       | description      |                  |                         |
+
+
+## Mandatory Header
+
+The first row MUST contain header columns using reserved names in the following order:
+
+* `gene_set_name`
+* `gene_set_description`
+* `gene_symbol`
+* `gene_description`
+
+Users MAY include additional header columns. Once gene sets are editted from cellxgene, the additional columns will no longer be stored in the file designated using the command --gene-sets-file.
+
+## Rows
+
+Each subsequent row describes a gene in a gene set.
+
+---
+
+### `gene_set_name`
+
+The `gene_set_name` column MUST contain a value. The values for `gene_set_name` MUST NOT contain illegal ASCII characters or sequences. If the following cases are detected, validation MUST display an error message and fail the upload:
```

Comment:
What does "fail the upload" and "validation" imply?   AFAIK, there is no "upload" in cellxgene, and it makes it sound like validation is a separate procedural step (if so, need to introduce it).

Instead of the above, if this is a data format specification, just state what is legal/illegal, and leave the UI/workflow out of it.  If it is a user guide, you will need to explain a bit more.